SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -204,6 +208,16 @@ void USpatialSender::CreateServerWorkerEntity(int AttemptCounter)
 		if (Op.status_code == WORKER_STATUS_CODE_SUCCESS)
 		{
 			Sender->NetDriver->WorkerEntityId = Op.entity_id;
+
+			// This logic is necessary because a potential race between the order in which the StartupActorManager
+			// AddComponentOp and ServerWorkerEntity CreateEntityResponseOp are received. The justification for this
+			// functionality more broadly is in UGlobalStateManager::ApplyStartupActorManagerData.
+			UGlobalStateManager* GlobalStateManager = Sender->NetDriver->GlobalStateManager;
```

Comment:
nit: I'd prefer a single function that you call here, that internally does all the should-id/auth etc checks. And just call it regardless from here, and from the Startup component add. Something like `AttemptSendWorkerReady`

Reply:
And then remove the additional gsm functions.